include/clap/helpers/plugin.hxx

```diff
@@ -791,6 +800,69 @@ namespace clap { namespace helpers {
       return self.paramsCount();
    }
 
+   //--------------------------------------//
+   // clap_plugin_configurable_audio_ports //
+   //--------------------------------------//
+
+   template <MisbehaviourHandler h, CheckingLevel l>
+   bool Plugin<h, l>::clapConfigurableAudioPortsCanApplyConfiguration(
+      const clap_plugin_t *plugin,
+      const clap_audio_port_configuration_request *requests,
+      uint32_t request_count) noexcept {
+      auto &self = from(plugin);
+      auto methodName = "clap_plugin_configurable_audio_ports.can_apply_configuration";
+      self.ensureMainThread(methodName);
+      self.ensureIsInactive(methodName);
+      self.ensureClapAudioPortConfigurationRequestIsValid(requests, request_count);
+
+      return self.configurableAudioPortsCanApplyConfiguration(requests, request_count);
+   }
+
+   template <MisbehaviourHandler h, CheckingLevel l>
+   bool Plugin<h, l>::clapConfigurableAudioPortsApplyActivation(
+      const clap_plugin_t *plugin,
+      const clap_audio_port_configuration_request *requests,
+      uint32_t request_count) noexcept {
+      auto &self = from(plugin);
+      auto methodName = "clap_plugin_configurable_audio_ports.apply_configuration";
+      self.ensureMainThread(methodName);
+      self.ensureIsInactive(methodName);
+      self.ensureClapAudioPortConfigurationRequestIsValid(requests, request_count);
+
+      if (l >= CheckingLevel::Minimal &&
+          !self.configurableAudioPortsCanApplyConfiguration(requests, request_count)) {
+         self.hostMisbehaving(
```

Comment:
From the spec it isn't mandatory to for the host to call `can_apply_configuration()` before calling `apply_configuration()`.

So your check is invalid.

Here what the helper should do:
1. call check can apply, and remember the value
2. forward the the apply call
3. check if can apply and apply returned the same value, if not, then the **plugin** is mis-behaving.

Reply:
Alright, requested changes are up via cccc104dc69502072b9fcf5528ae1fbed3bffb55